src/core/qgsinterval.h

```diff
@@ -187,6 +187,35 @@ class CORE_EXPORT QgsInterval
      */
     void setValid( bool valid ) { mValid = valid; }
 
+    /**
+     * Returns the interval duration.
+     *
+     * The interval duration is set only through the
```

Comment:
I don't think this is so nice -- the other constructors could still set the unit and duration using seconds, or in the case of the y/m/d/h/min/sec constructor we could check to see if only one of the units is non zero and then use that as the duration/unit (falling back to seconds, and appropriately documenting this)

I'd also suggest we rename these to something like "originalDuration" and "originalUnit", and expand on the dox to explain how they differ from the other existing methods.

Reply:
ok,

> constructor we could check to see if only one of the units is non zero and then use that as the duration/unit

I haven't quite get you here.

 I'm thinking of adding the following changes into the constructors, is this ok?

```cpp
QgsInterval::QgsInterval( double seconds )
  : mSeconds( seconds )
  , mValid( true )
{
   mOriginalDuration = mSeconds;
   mOriginalUnit = QgsUnitTypes::TemporalSeconds;
}

QgsInterval::QgsInterval( double years, double months, double weeks, double days, double hours, double minutes, double seconds )
  : mSeconds( years * QgsUnitTypes::fromUnitToUnitFactor( QgsUnitTypes::TemporalYears, QgsUnitTypes::TemporalSeconds )
              + months * QgsUnitTypes::fromUnitToUnitFactor( QgsUnitTypes::TemporalMonths, QgsUnitTypes::TemporalSeconds )
              + weeks * QgsUnitTypes::fromUnitToUnitFactor( QgsUnitTypes::TemporalWeeks, QgsUnitTypes::TemporalSeconds )
              + days * QgsUnitTypes::fromUnitToUnitFactor( QgsUnitTypes::TemporalDays, QgsUnitTypes::TemporalSeconds )
              + hours * QgsUnitTypes::fromUnitToUnitFactor( QgsUnitTypes::TemporalHours, QgsUnitTypes::TemporalSeconds )
              + minutes * QgsUnitTypes::fromUnitToUnitFactor( QgsUnitTypes::TemporalMinutes, QgsUnitTypes::TemporalSeconds )
              + seconds )
  , mValid( true )
{
    mOriginalDuration = mSeconds;
    mOriginalUnit = QgsUnitTypes::TemporalSeconds;
}
```

Reply:
What I mean is something like this:

```
QgsInterval::QgsInterval( double years, double months, double weeks, double days, double hours, double minutes, double seconds )
  : mSeconds( years * QgsUnitTypes::fromUnitToUnitFactor( QgsUnitTypes::TemporalYears, QgsUnitTypes::TemporalSeconds )
              + months * QgsUnitTypes::fromUnitToUnitFactor( QgsUnitTypes::TemporalMonths, QgsUnitTypes::TemporalSeconds )
              + weeks * QgsUnitTypes::fromUnitToUnitFactor( QgsUnitTypes::TemporalWeeks, QgsUnitTypes::TemporalSeconds )
              + days * QgsUnitTypes::fromUnitToUnitFactor( QgsUnitTypes::TemporalDays, QgsUnitTypes::TemporalSeconds )
              + hours * QgsUnitTypes::fromUnitToUnitFactor( QgsUnitTypes::TemporalHours, QgsUnitTypes::TemporalSeconds )
              + minutes * QgsUnitTypes::fromUnitToUnitFactor( QgsUnitTypes::TemporalMinutes, QgsUnitTypes::TemporalSeconds )
              + seconds )
  , mValid( true )
{
   if ( years && !months  && !weeks  && !days && !hours && !minutes && !seconds  )
   {
      mOriginalDuration =years;
      mOriginalUnit = QgsUnitTypes::TemporalYears;
   }
   else if ( !years && months && !weeks  && !days && !hours && !minutes && !seconds  ) 
    {
      mOriginalDuration =months;
      mOriginalUnit = QgsUnitTypes::TemporalMonths;
      }
   else if ( !years && !months && weeks  && !days && !hours && !minutes && !seconds  ) 
    {
      mOriginalDuration =weeks;
      mOriginalUnit = QgsUnitTypes::TemporalWeeks;
     }
     .. .etc ...
    else
    {
       mOriginalUnit = QgsUnitTypes::TemporalUnknownUnit;
    }

}

```

Reply:
ok I see, what about the constructor which takes only seconds should I also change it the way I have suggested?

Reply:
Yep that looks good!